pkg/apiclient/client_factory.go

```diff
@@ -21,11 +21,11 @@ import (
 type ClientFactory interface {
 	// GetSpacedClient returns an Octopus api Client instance which is bound to the Space
 	// specified in the OCTOPUS_SPACE environment variable, or the command line. It should be the default
-	GetSpacedClient() (*octopusApiClient.Client, error)
+	GetSpacedClient(requester Requester) (*octopusApiClient.Client, error)
```

Comment:
Given that `Requester` is basically just `func () string` perhaps the code would be simpler and easier to understand if the signature changed to `GetSpacedClient(userAgent func() string)`? When I saw the 'Requester' interface I assumed some complex networking behaviour was going on, and I had to spend the time to figure it out. `func() string` would have made this more obvious.

You wouldn't need the ceremony with the fakeRequesterContext either

Reply:
Thanks @borland, agreed that this could be an inline func, but I am going to leave it as it for the time being.